vmm/src/vm.rs

```diff
@@ -481,16 +483,30 @@ impl Vm {
         Ok(())
     }
 
-    pub fn resize(&mut self, desired_vcpus: u8) -> Result<()> {
-        self.cpu_manager
-            .lock()
-            .unwrap()
-            .resize(desired_vcpus)
-            .map_err(Error::CpuManager)?;
-        self.devices
-            .notify_hotplug(HotPlugNotificationType::CPUDevicesChanged)
-            .map_err(Error::DeviceManager)?;
-        self.config.lock().unwrap().cpus.boot_vcpus = desired_vcpus;
+    pub fn resize(&mut self, desired_vcpus: Option<u8>, desired_memory: Option<u64>) -> Result<()> {
+        if let Some(desired_vcpus) = desired_vcpus {
+            self.cpu_manager
+                .lock()
+                .unwrap()
+                .resize(desired_vcpus)
+                .map_err(Error::CpuManager)?;
+            self.devices
+                .notify_hotplug(HotPlugNotificationFlags::CPU_DEVICES_CHANGED)
+                .map_err(Error::DeviceManager)?;
```

Comment:
We may want to roll back here, i.e resize back to the previous number of vcpus when we can't notify the guest about them.
This would be material for a follow up PR, obviously.